src/Nethermind/Nethermind.Blockchain/Producers/MultipleBlockProducer.cs

```diff
@@ -0,0 +1,132 @@
+//  Copyright (c) 2021 Demerzel Solutions Limited
+//  This file is part of the Nethermind library.
+// 
+//  The Nethermind library is free software: you can redistribute it and/or modify
+//  it under the terms of the GNU Lesser General Public License as published by
+//  the Free Software Foundation, either version 3 of the License, or
+//  (at your option) any later version.
+// 
+//  The Nethermind library is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+//  GNU Lesser General Public License for more details.
+// 
+//  You should have received a copy of the GNU Lesser General Public License
+//  along with the Nethermind. If not, see <http://www.gnu.org/licenses/>.
+// 
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.FSharp.Core;
+using Nethermind.Consensus;
+using Nethermind.Core;
+
+namespace Nethermind.Blockchain.Producers
+{
+    public abstract class MultipleBlockProducer<T> : IBlockProducer where T : IBlockProducerInfo
+    {
+        private readonly IBlockProductionTrigger _blockProductionTrigger;
+        private readonly IBestBlockPicker _bestBlockPicker;
+        private readonly T[] _blockProducers;
+
+        protected MultipleBlockProducer(
+            IBlockProductionTrigger blockProductionTrigger,
```

Comment:
Move to MEV plugin unless used in kther places

Reply:
i wouldnt, im going to have to use it in Account Abstraction for example